src/main/java/io/gravitee/common/utils/RxHelper.java

```diff
@@ -37,4 +43,67 @@ public class RxHelper {
     }
+
+    /**
+     * Returns a {@link FlowableTransformer} that can be used in a composition.
+     * It applies an interval for each element received in the {@link Flowable} upstream in opposition to {@link Flowable#delay(long, TimeUnit)} which will apply the delay only once for the {@link Flowable}
+     *
+     * @param delay the delay to apply between each element of the {@link Flowable}
+     * @param timeUnit the {@link TimeUnit}  of the delay
+     *
+     * @return a {@link FlowableTransformer} that will be applied.
+     */
+    public static <R> FlowableTransformer<R, R> delayElement(int delay, TimeUnit timeUnit) {
+        return upstream -> upstream.zipWith(Flowable.interval(delay, timeUnit), (item, interval) -> item);
+    }
+
+    /**
+     * Returns a {@link FlowableTransformer} that can be used in a composition.
+     * It retries the {@link Flowable} X times with delay between each attempt
+     *
+     * @param times the attempts number
+     * @param backOffDelay the delay between each retry
+     * @param timeUnit the {@link TimeUnit} of the backOffDelay
+     *
+     * @return a {@link FlowableTransformer} that will be applied.
+     */
+    public static <R> FlowableTransformer<R, R> retryFlowable(int times, int backOffDelay, TimeUnit timeUnit) {
+        return upstream ->
+            upstream.retryWhen(throwables -> throwables.compose(delayElement(backOffDelay, timeUnit)).compose(takeThenThrow(times)));
+    }
+
+    /**
+     * Returns a {@link CompletableTransformer} that can be used in a composition.
+     * It retries the {@link Flowable} X times with delay between each attempt
+     *
+     * @param times the attempts number
+     * @param backOffDelay the delay between each retry
+     * @param timeUnit the {@link TimeUnit} of the backOffDelay
+     *
+     * @return a {@link CompletableTransformer} that will be applied.
+     */
+    public static CompletableTransformer retry(int times, int backOffDelay, TimeUnit timeUnit) {
+        return upstream ->
+            upstream.retryWhen(throwables -> throwables.compose(delayElement(backOffDelay, timeUnit)).compose(takeThenThrow(times)));
+    }
+
+    /**
+     * Returns a {@link FlowableTransformer} that can be used in a composition.
+     * It ignores the N first throwable then return the N+1 in error.
+     *
+     * @param limit the number of throwables to ignore
+     *
+     * @return a {@link FlowableTransformer} that will be applied.
+     */
+    private static <R extends Throwable> FlowableTransformer<R, R> takeThenThrow(final int limit) {
+        final AtomicInteger tries = new AtomicInteger(0);
+        return upstream ->
+            upstream.flatMapMaybe(throwable -> {
+                if (tries.incrementAndGet() > limit) {
+                    return Maybe.error(throwable);
+                } else {
+                    return Maybe.just(throwable);
```

Comment:
Maybe.empty ?

Reply:
I have to keep it because the calling `retryWhen` will retry only of it receives a throwable.